src/app/core/services/misc/github-client.service.ts

```diff
@@ -0,0 +1,39 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http'
+import { Injectable } from '@angular/core'
+
+import { DefaultGithubToken } from '../../../../assets/data/defaultConfig'
+import { ConfigKeys } from '../../../shared/enums/config'
+import { GithubContent } from '../../../shared/models/github'
+import { RemoteConfigService } from '../config/remote-config.service'
+
+@Injectable()
+export class GithubClientService {
+  constructor(
+    private http: HttpClient,
+    private remoteConfig: RemoteConfigService
+  ) {}
+
+  get(url): Promise<any> {
+    return this.remoteConfig
+      .read()
+      .then(config =>
+        config.getOrDefault(ConfigKeys.GITHUB_API_TOKEN, DefaultGithubToken)
+      )
+      .then(token =>
+        this.http
+          .get(url, {
+            headers: new HttpHeaders().set('Authorization', 'Bearer ' + token)
```

Comment:
Can you make this so that if the token is not supplied or is empty it does not use auth and try to get the questionnaires without any auth (this would work when the repo is public, so no need to supply the token)

Reply:
Yeah also just realized that and added it. Will commit.